src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx

```diff
@@ -813,6 +814,30 @@ Currently, in order to export a File PFB, \`enableLimitedFilePFBExport\` must be
     );
   };
 
+  isLoginForButtonFeatureEnabled = () => {
+    if (this.props.buttonConfig.loginForButtonFeature) {
+      return this.goToLogin();
+    }
+    return false;
+  }
+
+  goToLogin = () => {
+    if (!this.props.user || !this.props.user.username) {
+      this.props.history.push('/login');
+      return true;
+    }
+    return false;
+  }
+
+  isDownloadButton = (buttonConfig) => {
+    if ((buttonConfig.type.startsWith('data')
+    || buttonConfig.type === 'manifest'
+    || buttonConfig.type === 'file-manifest')) {
+      return true;
+    }
+    return false;
+  }
+
   render() {
     const dropdownConfigs = calculateDropdownButtonConfigs(this.props.buttonConfig);
```

Comment:
we should apply the same logic to dropdown buttons as well right?

Reply:
I am not sure about this.

Reply:
Which part is confusing you?

`ExplorerButtonGroup` can have two types of buttons, regular button and dropdown buttons. What you are changing now will not affect dropdown buttons, because they are not generated by calling `renderButton()`

So `loginForButtonFeature` will not work on dropdown buttons. MIDRC currently don't have any of them bit other envs do. So it would be ideal if all download buttons work in the same way

Or at least make a new ticket if you think this requirement is out of scope of your ticket